sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -175,7 +176,9 @@ abstract class JdbcDialect extends Serializable with Logging{
   def compileValue(value: Any): Any = value match {
     case stringValue: String => s"'${escapeSql(stringValue)}'"
     case timestampValue: Timestamp => "'" + timestampValue + "'"
+    case timestampValue: Instant => "'" + timestampValue + "'"
```

Comment:
I would guess that the default `Instant` formatting by `toString()` might be not accepted by dbms. Could you take a look at `FractionTimestampFormatter`: https://github.com/apache/spark/blob/3a299aa6480ac22501512cd0310d31a441d7dfdc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala#L121

Reply:
I added the TimestampFormatter. I saw there's a DateFormatter as well. Should I use it for LocalDate or is toString() ok?